src/main/java/com/suchocki/parkingmeter/dao/ParkActionDAOImpl.java

```diff
@@ -0,0 +1,72 @@
+package com.suchocki.parkingmeter.dao;
+
+import java.util.List;
+
+import org.springframework.stereotype.Repository;
+import com.suchocki.parkingmeter.database.FakeDatabaseStub;
+import com.suchocki.parkingmeter.entity.Driver;
+import com.suchocki.parkingmeter.entity.ParkAction;
+
+@Repository
+public class ParkActionDAOImpl implements ParkActionDAO {
+
+	@Override
+	public void save(ParkAction parkAction) {
+		for (ParkAction pAction : FakeDatabaseStub.parkActions) {
+			if (pAction.getId() == parkAction.getId()) {
+				update(parkAction);
+			}
+		}
+		parkAction.setId(FakeDatabaseStub.parkActions.size() + 1);
```

Comment:
a jak `parkAction` będzie już miało `id` bo to `update` to mu je nadpiszesz?

Reply:
Oczywiście, po _update_, powinien być _return;_.